### PR TITLE
Make tests less specific about feature equality

### PR DESCRIPTION
**Pull request by sgillies**
Also reformat.

Towards resolving #935 #1031

### CODE REVIEW

File: tests/test_drvsupport.py [hunk 1]
```diff
@@ -10,94 +10,104 @@
...
             open_driver = driver
         else:
             open_driver = None
-        with fiona.open(path, driver=open_driver) as c:
-            assert c.driver == driver
-            items = list(c)
-            assert len(items) == len(records1)
-            for val_in, val_out in zip(records1, items):
-                assert test_equal(driver, val_in, val_out), "in: {val_in}, out: {val_out}".format(val_in=val_in,
-                                                                                                  val_out=val_out)
```
Comment (sgillies) [inline tests/test_drvsupport.py:64]: #935 and #1031 report test failures involving coordinate comparison. So I'm removing the feature comparisons. We are really only concerned about the counting of features here. The exact values can be driver/platform dependent.

File: tests/test_drvsupport.py [hunk 2]
```diff
@@ -225,73 +251,73 @@ def test_no_write_driver_cannot_write(tmpdir, driver, testdata_generator, monkey
...
-                    crs=crs,
-                    schema=schema,
-                    **create_kwargs) as c:
+    with fiona.open(
+        path, "w", driver=driver, crs=crs, schema=schema, **create_kwargs
+    ) as c:
 
         c.writerecords(records1)
 
-    is_good = True
```
Comment (sgillies) [inline tests/test_drvsupport.py:268]: This complicated the test and made it hard to understand. I've tried to rewrite it without this flag.

File: tests/test_drvsupport.py [hunk 3]
```diff
@@ -225,73 +251,73 @@ def test_no_write_driver_cannot_write(tmpdir, driver, testdata_generator, monkey
...
     try:
-        with fiona.open(path, 'a',
-                        driver=driver) as c:
+        with fiona.open(path, "a", driver=driver) as c:
             c.writerecords(records2)
+    except Exception as exc:
+        log.exception("Caught exception in trying to append.")
```
Comment (sgillies) [inline tests/test_drvsupport.py:307]: Reduced the scope of the try/except block.

File: tests/test_drvsupport.py [hunk 4]
```diff
@@ -225,73 +251,73 @@ def test_no_write_driver_cannot_write(tmpdir, driver, testdata_generator, monkey
...
-        is_good = False
+    if driver in {"FileGDB", "OpenFileGDB"}:
+        open_driver = driver
+    else:
+        open_driver = None
 
-    assert not is_good
+    with fiona.open(path, driver=open_driver) as collection:
+        assert collection.driver == driver
+        assert len(collection) == len(records1)
```
Comment (sgillies) [inline tests/test_drvsupport.py:316]: I think this is the crux of it. If no new records were added and there was no other exception, the format does not support appending.

**Comment by sgillies** [comment]
PCIDSK is a difficult vector format. It doesn't appear in https://gdal.org/drivers/vector/index.html and got a rewrite at 3.3.0. The driver in versions back to 2.1 seems to support append, at least in my more general rewrite of the tests, so I've adjusted the metadata.

**Comment by mwtoews** [comment]
Re PCIDSK, see https://github.com/Toblerity/Fiona/pull/1046#issuecomment-974906111 which was put into the master branch. I recall looking at the release notes for that GDAL release, but didn't see much for PCIDSK.

**Pull request merged**